### PR TITLE
refactor: delete manual interfaces for changelogs

### LINKED ISSUE #1719: Remove `CHANGELOG.md`
# Feature Request

With the ability to auto-generate changelogs during a release in Github, and the usage of conventional commits, we don't need to manually create and update a Changelog anymore.
Thus, the `CHANGELOG.md` file can be deleted, and the related workflow to check the changelog can go away too.

## Which Areas Would Be Affected?
Build

## Why Is the Feature Desired?
Avoids pesky merge conflicts on almost _every_ commit.

## Solution Proposal
- delete `CHANGELOG.md`
- delete the `check-for-changelog-modifications` job in the `scan-pull-request.yaml` workflow
- delete the "added relevant details...." section from `PULL_REQUEST_TEMPLATE.md`
- remove the relevant section from `CONTRIBUTING.md`

## Type of Issue
improvement

## Checklist

- [ ] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

### PR DESCRIPTION
**Pull request by janpmeyer**
## What this PR changes/adds

In this pull request all manual interfaces for changelog are removed.

- delete CHANGELOG.md
- delete the check-for-changelog-modifications job in the scan-pull-request.yaml workflow
- delete the "added relevant details...." section from PULL_REQUEST_TEMPLATE.md
- delete the "update changelog..." from release.md
- remove the relevant section from CONTRIBUTING.md

## Why it does that

Changelogs will be from now on auto generated.

## Linked Issue(s)

Closes #1719

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [x] formatted title correctly?

### CODE REVIEW

**Review by ndr-brt — APPROVED** [review]
Great, I noticed also these two lines to be deleted from the `release.md` file:
https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/docs/developer/releases.md?plain=1#L152-L153

**Comment by janpmeyer** [comment]
> Great, I noticed also these two lines to be deleted from the `release.md` file: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/docs/developer/releases.md?plain=1#L152-L153

Added in the newest commit.

**Review by ndr-brt — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -22,5 +22,4 @@ Closes # <-- _insert Issue number if one exists_
 - [ ] added/updated copyright headers?
 - [ ] documented public classes/methods?
 - [ ] added/updated relevant documentation?
-- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
```
Comment (paullatzelsperger) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: instead of completely removing this section I would add a bullet point:
`- [ ] assigned correct labels (exclude from changelog with label "no-changelog")?`
 
A bit of background: due to the auto-generating of Changelogs, PRs must now have the correct labels as well, so that they get categorized correctly. If no label is assigned, the PR will get categorized under `Features & Improvements`. `bug` and `documentation` should be self-explanatory.

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (janpmeyer) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Added your suggestion into the template.

Comment (juliapampus) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: @paullatzelsperger Maybe self-explanatory but now known that this will be filtered. Do we have an explanation anywhere? Otherwise, I would rephrase to:
```
- [ ] assigned correct labels? (`bug`, `documentation`, exclude from changelog with `no-changelog`)
```

Comment (juliapampus) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: I think so, but using one of those 3 labels is not mandatory. Every other PR will also end up in the changelog. Does it make sense to force people to use at least one label? @ndr-brt

Comment (ndr-brt) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: @juliapampus my only concern is to "forget" to set the label and have a lot of PRs going to the wrong section by default (I tend to forget about checklist after some PRs). Having it mandatory will make everyone to have to think about what's the right section, but I agree on the fact that could be bothersome.

Comment (juliapampus) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: @ndr-brt @paullatzelsperger Would something like #1737 help?

Comment (juliapampus) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: That would also allow to filter the automated changelog in more detail

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -22,5 +22,5 @@ Closes # <-- _insert Issue number if one exists_
...
+- [ ] assigned correct labels? (exclude from changelog with label "no-changelog")
```
Comment (juliapampus) [inline .github/PULL_REQUEST_TEMPLATE.md:25]:
```suggestion
- [ ] assigned correct labels? (exclude from changelog with label `no-changelog`)
```

**Review by juliapampus — CHANGES_REQUESTED** [review]

**Review by juliapampus — APPROVED** [review]
